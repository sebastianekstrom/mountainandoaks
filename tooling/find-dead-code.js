const { exec } = require("child_process");

const findDeadCode = () => {
  exec("ts-prune | wc -l", (error, stdout) => {
    const amount = parseInt(stdout.trim(), 10);

    if (error) {
      console.log(error);
      return process.exit(1);
    }

    if (amount !== 0) {
      const pluralized = amount === 1 ? "instance" : "instances";
      console.log(
        `ðŸš¨ ${amount} ${pluralized} of dead code found. Run 'ts-prune' for more details`
      );
      return process.exit(1);
    }

    console.log("âœ… No instances of dead code found");
  });
};

findDeadCode();
