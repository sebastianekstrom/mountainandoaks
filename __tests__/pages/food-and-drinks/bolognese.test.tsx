import React from "react";
import { render, screen } from "../../../tooling/tests";
import Bolognese from "pages/food-and-drinks/bolognese";

describe("Bolognese page", () => {
  it("renders title correctly", () => {
    render(<Bolognese />);
    const title = screen.queryAllByText(/Bolognese/i);
    expect(title).toBeDefined();
  });

  it("renders the ingress text", () => {
    render(<Bolognese />);
    const ingress = screen.queryAllByText(
      /There are almost as many Bolognese recipes as there are Italians/i
    );
    expect(ingress).toBeDefined();
  });

  it("renders the ingredients", () => {
    render(<Bolognese />);
    const ingredient = screen.queryAllByText(/Minced beef/i);
    expect(ingredient).toBeDefined();
  });

  it("renders the method steps", () => {
    render(<Bolognese />);
    const step = screen.queryAllByText(/Start off with some preparation/i);
    expect(step).toBeDefined();
  });
});
