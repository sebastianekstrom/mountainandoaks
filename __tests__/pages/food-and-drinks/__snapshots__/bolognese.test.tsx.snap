// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolognese page renders the correct ingredients 1`] = `
<section
  class="mb-8 lg:mb-0"
  data-testid="INGREDIENT_LIST"
>
  <div>
     
  </div>
  <div
    class="mt-8"
  >
    <div
      class="mb-2"
    >
      <h3
        class="variable font-heading text-h3 leading-h3 text-regular "
      >
        Bolognese
      </h3>
    </div>
    <div
      class="flex flex-col"
    >
      <ul>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Minced beef
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            500
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Chicken stock
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            500
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              White wine
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            230
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Milk
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            230
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Bacon or pancetta
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            100
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Tomato paste
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            75
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Onion
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            70
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Celery
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            35
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Carrot
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            35
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Demi-glace (optional)
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            30
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Cloves of garlic
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            0.8
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Bay leaf
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            1
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Pinch of dried oregano
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          />
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Pinch of nut meg
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          />
        </li>
      </ul>
    </div>
  </div>
  <div
    class="mt-8"
  >
    <div
      class="mb-2"
    >
      <h3
        class="variable font-heading text-h3 leading-h3 text-regular "
      >
        Pasta
      </h3>
    </div>
    <div
      class="flex flex-col"
    >
      <ul>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Spaghetti or linguine
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            500
            g
          </p>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="mt-8"
  >
    <div
      class="mb-2"
    >
      <h3
        class="variable font-heading text-h3 leading-h3 text-regular "
      >
        Garnish
      </h3>
    </div>
    <div
      class="flex flex-col"
    >
      <ul>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Grated Parmesan
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          />
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Splash of olive oil
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          />
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Freshly ground black pepper
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          />
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Fresh oregano
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          />
        </li>
      </ul>
    </div>
  </div>
</section>
`;
